### DISCUSSION TITLE
Distributed RabbitMQ  Federated  vs Shovel

### DISCUSSION THREAD

**Discussion by iamnhphu** [rabbitmq/rabbitmq-server#10504] [Questions]
Hello everyone.

![image](https://github.com/rabbitmq/rabbitmq-server/assets/37231397/d63be029-d3c4-48ec-a103-769b74caa738)

I need to send a message from server to a specific client. Client inside a restricted environment, can only make http request. Therefore, I use long polling to get message.

App #1, App #2, App #3 is using to keep connection with rabbitmq server and response message to client if there's a message.

Each RabbitMQ instance in different zone (less than 100km), Clustering can't be used. 
This is first time I use message broker, so I don't know how to choose between Federation exchange / Federation queue / Static Shovels / Dynamic Shovels.

read through the documents, it seems that Shovels will move message to downstream if there's no consumer.
Message will be missed or late if long polling timeout and reconnect to upstream at the same time message move to downstream, correct?

Federated Exchange: all nodes will receive a copy of messages; I don't want a message to be send multiple time. Is "exactly one delivery" possible?

Is Federated Queue best solution for me? Would run each node in Non clustered mode, and each App open 3 connection to 3 rabbitmq server better?

Thanks everyone.

**Comment by michaelklishin** [comment]
I don't see why these three nodes cannot [be a cluster](https://www.rabbitmq.com/clustering.html) without any shovels or queue federation unless you are **certain** you need them and can explain why.

**Reply by iamnhphu** [reply]
[https://blog.rabbitmq.com/posts/2020/07/disaster-recovery-and-high-availability-101/](url)
this blog post says that "Clustering requires highly reliable, low latency links between brokers. Clustering over a WAN is highly discouraged." That is why I don't consider it.
